src/types.js

```diff
@@ -2,18 +2,53 @@
 
 import { type Schema } from "normalizr";
 
+export type ReferencesToType = {
+  [toEntity: string]: {
+    [from: string]: null
+  }
+};
+
 export type SchemaMapType<SchemasType: string> = {
   [schemaName: SchemasType]: Schema
 };
 
-export type EntityType = Object;
+export type EntityType = { id: string };
 
 export type SchemaEntitiesMapType = {
   [id: string]: EntityType
 };
 
+type RelationTypeType = "one" | "many";
+
+export type SchemaReferenceType = {
+  toSchema: string,
+  viaField: string,
+  relationType: RelationTypeType
+};
+
+export type EntityReferenceType = {
+  fromSchema: string,
+  fromID: string,
+  viaField: string,
+  relationType: RelationTypeType
+};
+
+export type SchemaReferencesType = {
+  [referencingSchema: string]: SchemaReferenceType[]
+};
+
+type ByIDMapType<D> = {
+  [id: string]: D
+};
+
+type BySchemaByIDMapType<D> = {
```

Comment:
Yes and no – it never needed to be dynamic. 
However, I believe it's more expressive for it to be a generic since it implies a clear focus of what is typed (a "by id"-map does not care for what's inside)
I refined to `BySchemaByIDMapType<EntityType>` though.